src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Max.cs

```diff
@@ -74,73 +75,64 @@ public static void Max<T>(ReadOnlySpan<T> x, ReadOnlySpan<T> y, Span<T> destinat
         /// </remarks>
         public static void Max<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
             where T : INumber<T> =>
-            InvokeSpanScalarIntoSpan<T, MaxPropagateNaNOperator<T>>(x, y, destination);
+            InvokeSpanScalarIntoSpan<T, MaxOperator<T>>(x, y, destination);
 
-        /// <summary>T.Max(x, y) (but NaNs may not be propagated)</summary>
-        internal readonly struct MaxOperator<T> : IAggregationOperator<T> where T : INumber<T>
+        /// <summary>Max(x, y)</summary>
+        internal readonly struct MaxOperator<T> : IAggregationOperator<T>
+             where T : INumber<T>
         {
             public static bool Vectorizable => true;
 
-            public static T Invoke(T x, T y)
-            {
-                if (typeof(T) == typeof(Half) || typeof(T) == typeof(float) || typeof(T) == typeof(double))
-                {
-                    return x == y ?
-                        (IsNegative(x) ? y : x) :
-                        (y > x ? y : x);
-                }
-
-                return T.Max(x, y);
-            }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T Invoke(T x, T y) => T.Max(x, y);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
-                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+#if !NET9_0_OR_GREATER
+                if ((typeof(T) == typeof(float)) || (typeof(T) == typeof(double)))
                 {
-                    if (AdvSimd.IsSupported && typeof(T) == typeof(float))
-                    {
-                        return AdvSimd.Max(x.AsSingle(), y.AsSingle()).As<float, T>();
-                    }
-
-                    if (AdvSimd.Arm64.IsSupported && typeof(T) == typeof(double))
-                    {
-                        return AdvSimd.Arm64.Max(x.AsDouble(), y.AsDouble()).As<double, T>();
-                    }
-
-                    return
-                        Vector128.ConditionalSelect(Vector128.Equals(x, y),
-                            Vector128.ConditionalSelect(IsNegative(x), y, x),
-                            Vector128.Max(x, y));
+                    return Vector128.ConditionalSelect(
+                        Vector128.LessThan(y, x) | IsNaN(x) | (Vector128.Equals(x, y) & IsNegative(y)),
+                        x,
+                        y
+                    );
                 }
+#endif
 
                 return Vector128.Max(x, y);
             }
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> y)
             {
-                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+#if !NET9_0_OR_GREATER
```

Comment:
Does this mean that .NET 9 includes a behavioral change around how Max handles floating point?

Reply:
Not for TensorPrimitives, yes for `Vector128/256/512`. 

`Vector128/256/512` were non-deterministic across hardware and `Max(x, y)` vs `Max(y, x)` could even return different results for some inputs (such as `NaN` or `+0 vs -0`). As of this PR, they are deterministic and exactly match the scalar APIs (and the TensorPrimitives APIs).